### ISSUE TITLE
[Bug]: Removing object from array not working?

### ISSUE THREAD

**Issue by Navil** [realm/realm-dart#914]
### What happened?

I have following code:

```
bookmarkUser(User user) {
    realm.write(() {
      if (user.likedUsers.contains(user)) {
        user.likedUsers.remove(user); //Returns true, but does not update realm
      } else {
        user.likedUsers.add(user); //This works the way it should
      }
    });
}
```

As mentioned in the comments, I am not able to remove elements from the array. The .remove function is called and true is returned. I also tried removeWhere, but with the same result - The data in realm is unchanged. 

I have a trigger on likedUsers, and that trigger also does not trigger (this is my indication that there is indeed nothing happening). No error or anything similar is printed.

### Repro steps

1.) Use a list of Objects in a model
2.) Add elements to that list
3.) Call list.remove(element)
4.) The user is still in the list

### Version

Channel stable, 3.3.2, on Microsoft Windows [Version 10.0.19043.2006], locale de-AT

### What Realm SDK flavor are you using?

MongoDB Atlas (i.e. Sync, auth, functions)

### What type of application is this?

Flutter Application

### Client OS and version

0.4.0+beta

### Code snippets

```
bookmarkUser(User user) {
    realm.write(() {
      if (user.likedUsers.contains(user)) {
        user.likedUsers.remove(user); //Returns true, but does not update realm
      } else {
        user.likedUsers.add(user); //This works the way it should
      }
    });
}
```

```
class _User{
  @PrimaryKey()
  @MapTo("_id")
  late ObjectId id;
  ...
  List<_User> likedUsers = [];
}
```

**Comment by blagoev** [comment]
What is your expectation the `user` is `removed` from the realm or the user is `deleted` from the realm. Remove only `removes` objects from the list, but the object still exists in the realm. For that you should use `realm.delete`

**Comment by Navil** [comment]
I want to remove the user from that list only.

Basically un-bookmark that user.

**Comment by blagoev** [comment]
Ok, I investigated a bit and it seems we indeed had a bug there which was fixed in https://github.com/realm/realm-dart/pull/884
Unfortunately we still have not released this fix. This will come sometime soon.

Thanks for reporting

**Comment by Navil** [comment]
Thank you very much. That explains why I found working tests in the repository and had the assumption that this should work as it is tested.

So I will have some patience. For those who also need a temporary workaround, I did the following:
```
removeElementFromRealmList<T extends RealmObject>(
      List<T> realmList, T element) {
    List<T> copy = List.from(realmList);
    copy.remove(element);
    realmList.clear();
    realmList.addAll(copy);
  }
```

**Comment by nielsenko** [comment]
@Navil We have just released 0.5.0+beta that includes the fix for this
